### PR TITLE
Is90a/dockerfile user

### LINKED ISSUE #90: (SimCoreUser) will be the new user for host&container in order to avoid being root. ID with a very high number. Mechanism that the user in the host and in docker image are the same one.
- [x] **scu** (SimCoreUser) will be the new user for host&container in order to avoid being root. ID with a very high number. Mechanism that the user in the host and in docker image are the same one. 
- [x] adapt director
- [x] adapt sidecar (connected to PR #199)
- [ ] remove the need of export RUN_DOCKER_ENGINE_ROOT=1 in Makefile and export DOCKER_GID=1042
export HOST_GID=1000

requires #215
links #300

### LINKED ISSUE #277: Update 3d-viewer docker to install simcore packages instead of git clone

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?
Continuation of PR #570 
Updates docker config for remaining services:
- [x] director: upgraded docker config (detects docker and host id and group), fixed hard-coded port, adapted non-std cli
- [x] sidecar: upgraded docker config(detects docker and host id and group)
- [x] storage: upgraded docker config(detects docker and host id and group), python2 and python3 installed in container's usr instead of venv
- [x] Fixes the case of using root host, as it happens in windows systems

## Related issue number
closes #90 
closes #277

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**

### CODE REVIEW

File: services/sidecar/docker/boot.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+
+# RUNNING application ----------------------------------------
+if [[ ${BOOT_MODE} == "debug" ]]
+then
+  # TODO: activate pdb??
+  DEBUG_LEVEL=debug
+  CONCURRENCY=1
+else
+  CONCURRENCY=2
```
Comment (mguidon) [inline services/sidecar/docker/boot.sh]: We should keep an eye on that. Concurrency should be done via scaling the sidecars. I just added the two for debugging

File: services/director/Dockerfile [hunk 2]
```diff
@@ -1,106 +1,143 @@
...
+
+RUN $MY_PIP install \
+      -r services/director/requirements/base.txt
 
-COPY --chown=scu:scu services/director/docker docker
 
 # --------------------------Development stage -------------------
+# Source code accessible in host but runs in container
+# Runs as myu with same gid/uid as host
+# Placed at the end to speed-up the build if images targeting production
```
Comment (sanderegg) [inline services/director/Dockerfile]: does not seem to be at the end

File: services/director/Dockerfile [hunk 3]
```diff
@@ -1,106 +1,143 @@
...
+ENV MY_BOOT_MODE development
+
 
-# in dev-mode we give access to `scu` to host's mapped volumes
-# FIXME: files created by scu cannot be deleted by host! we need to do the same group in host?
-RUN addgroup -g $HOST_GID_ARG hgrp &&\
-    addgroup scu hgrp && \
-    chown -R scu:scu $HOME/.venv &&\
-    rm -rf /tmp/.pipcache
+#  TODO: consider installing tooling in a separate virtualenv?
```
Comment (sanderegg) [inline services/director/Dockerfile]: what is the meaning of this?

File: services/storage/docker/boot.sh [hunk 4]
```diff
@@ -1,24 +1,44 @@
...
-  $PIP install -r requirements/dev.txt
-  $PIP list
+fi
 
-  cd $HOME/
-  simcore-service-storage --config docker-dev-config.yaml
-elif [[ ${DEBUG} == "2" ]]
+
+# RUNNING application ----------------------------------------
+if [[ ${BOOT_MODE} == "debug" ]]
```
Comment (mguidon) [inline services/storage/docker/boot.sh]: use MY_BOOT_MODE to be consistent

File: services/director/Dockerfile (hunk 3 above)
Comment (pcrespov) [reply services/director/Dockerfile]: The TODO actually do not belong here ... The idea is that sometimes we need tools (e.g. pip-compile or cookiecutter or linter) that "polute" with dependencies the py-environment where my app actually runs. Then it is desirable to isolate the env for the app and that of tooling

File: services/sidecar/docker/boot.sh (hunk 1 above)
Comment (pcrespov) [reply services/sidecar/docker/boot.sh]: I kept this as it was. I guess this means that every sidecar has actually TWO workers which is kind of wrong, provided that they share the same input/output folder!!! PLEASE comment on this.

File: Makefile [hunk 5]
```diff
@@ -115,7 +99,8 @@ rebuild-devel: .env .tmp-webclient-build
 
 .tmp-webclient-build:
 	# TODO: fixes having services_webclient:build present for services_webserver:production when
-	# targeting services_webserver:development
+	# targeting services_webserver:development and ensures source-output folder at host
+	mkdir -p services\web\client\source-output
```
Comment (odeimaiz) [inline Makefile]: This doesn't work with PowerShell. It's not wrong but maybe we should specify what kind of command-line shell to use.

Comment (pcrespov) [reply Makefile]: why don't u use [WSL] instead??? :-(

[WSL]:https://docs.microsoft.com/en-us/windows/wsl/install-win10

**Pull request merged**